src/scrapers/main.py

```diff
@@ -31,22 +36,34 @@ def docs():
         print('{: <16}{}'.format(s.code, s.name))
 
 
-(opts, args) = parser.parse_args()
+arguments = parser.parse_args()
 
-if opts.help:
+if arguments.help:
     docs()
 else:
-    print(f'Running {len(args)} scrapers. Use --help to see all individual scrapers')
+    print(f'Running {len(arguments.scrapers)} scrapers. Use --help to see all individual scrapers')
+    if arguments.group:
+        if arguments.group != InitiativeGroup.SUPPLY \
+                and arguments.group != InitiativeGroup.DEMAND:
+            print(f"Invalid group given {arguments.group}")
+            exit()
+
+    run_scrapers = None
+    if arguments.scrapers:
+        run_scrapers = reduce(lambda a, b: a + b, arguments.scrapers)
+
     for scraper in scrapers:
-        if opts.limit:
-            scraper.limit = opts.limit
+        if arguments.limit:
+            scraper.limit = arguments.limit
+        if arguments.group:
+            scraper.set_group(InitiativeGroup.DEMAND if arguments.group == "demand" else InitiativeGroup.SUPPLY)
 
         # preferably each scraper runs in it's own thread.
-        if not args:
+        if not run_scrapers:
             scraper.scrape()
-        elif args and scraper.code in args:
+        elif scraper.code in run_scrapers:
```

Comment:
Als dit de reden is van de reduce() naar een grote string dan kun je ook `if str(scraper.code) in run_scrapers` gebruiken

Reply:
oh wacht, het is al een string :D ok dan lijkt het overbodig

Reply:
;)